src/worker/generate-kmp.ts

```diff
@@ -0,0 +1,37 @@
+import { generateKmpJson } from "./generate-kmp-json";
+import { createZipWithFiles } from "./generate-zip";
+import {
+  compileModelFromLexicalModelSource,
+  WordListFromArray,
+} from "@predictive-text-studio/lexical-model-compiler";
+
+/**
+ * Give file information and create a kmp file for the dictionary
+ *
+ * Format of the lexical model file: {model_id}.model.js
+ * the compiled code generated by the @predictive-text-studio/lexical-model-compiler
+ *
+ * @param modelID
+ * @param compiledModelCode
+ * @param modelInfo
+ */
+export async function generateKmp(
+  langName: string,
+  bcp47Tag: string,
+  sources: WordListFromArray[],
+  modelId: string
+): Promise<ArrayBuffer> {
+  const kmpJsonFile = generateKmpJson({
+    languages: [{ name: langName, id: bcp47Tag }],
+  });
+  const modelFile = compileModelFromLexicalModelSource({
+    format: "trie-1.0",
+    sources: sources,
+  });
+
+  const kmpFile = createZipWithFiles({
+    [`${modelId}.model.js`]: modelFile,
+    "kmp.json": kmpJsonFile,
+  });
+  return kmpFile;
```

Comment:
**[Question]**: It seems like the linting is not doing the job here 😭, I will double check later

Reply:
What do you mean? I see absolutely nothing wrong with the code style here. Could you please clarify?

Reply:
There is inconsistency in the linting patter, for instance, there is no extra line after line 35. I wasn't sure about the linting rule we have and I thought that I might not implemented the husky hook correctly last time, which is why I put a "I will double check later here". Now that I am aware there is rule in our linting.

Reply:
It's irrelevant if the husky hook runs correctly, because the build literally fails if the linter dislikes 

For future reviews, please do not focus on code style. That's why we're running the linter on CI. And it [does work](https://github.com/eddieantonio/predictive-text-studio/runs/1196158848). If the linter accepts it, then it's acceptable code, as far as style is concerned. Please comment on structure, form, and meaning instead.

Reply:
Thank you again for feedback and clarifying the norm for pr review for the project. I am more than happy to take your feedback.